src/background/utils/extension-api.ts

```diff
@@ -66,32 +67,32 @@ export async function readLocalStorage<T extends {[key: string]: any}>(defaults:
 }
 
 export async function writeSyncStorage<T extends {[key: string]: any}>(values: T): Promise<void> {
-    return new Promise<void>((resolve, reject) => {
-        isWriting = true;
+    return new Promise<void>(async (resolve, reject) => {
+        await mutexStorageWriting.lock();
         chrome.storage.sync.set(values, () => {
             if (chrome.runtime.lastError) {
```

Comment:
In case of an error this code seems to never release the lock leading to a deadlock. Error might be temporary, but there is no mechanism for resolving the deadlock.

Reply:
Woops, yup correct. Good catch!